src/Forms/Validator.php

```diff
@@ -313,7 +313,10 @@ public static function validateInteger(Control $control): bool
 	 */
 	public static function validateFloat(Control $control): bool
 	{
-		$value = str_replace([' ', ','], ['', '.'], $control->getValue());
+		$value = $control->getValue();
+		if (is_string($value)) {
+			$value = str_replace([' ', ','], ['', '.'], $value);
```

Comment:
Has nothing to do with PR which is ok IMHO, but now saw this code...
Isn't this dangerous? We're just  replacing , with .? Are we sure , is not a thousand separator, in many regional settings it is?
@dg

Reply:
"," as thousand separator is an unfortunate choice, however 12.E3 is a valid float number but I don't think average people write numbers that way :-)

Float shouldn't be used to write numbers for the general case (it's better reserved for scientific purposes)

Reply:
That really depends from which part of world you're comming from and framework should never go in Europe/US direction here but supports both. Here is hard in some cases even impossile to determine what is
"1.234" or "1,234", both . and , can be either thousand separetor or decimal point...

Reply:
Imho it's safe to rely on browser behavior. `<input type="number" step="any">` allows you to write only `,` or `.`, never both. Also sent value is normalized to contain `.`, not `,`.

In case someone uses nette/forms outside of the html context... well, they just shouldn't.

Reply:
While I agree that number type is the way, unfortunately some people (especially accountants) want the thousand separator. Actually I use a text type with a custom validator for the format 1.234,56 like what they see in a desk calculator. 

Sometimes I agree that dealing with big numbers the thousand separator is useful, but this could be solved if the input type number may implement a thousand separator (without any javascript trick).

In my opinion nette/forms should treat numbers as string, for the same reason outlined here: https://github.com/nette/database/issues/257